src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -530,3 +543,32 @@ export const performEncryptPostSubmissionActions = (
       })
     })
 }
+
+export const getQuarantinePresignedPostData = (
+  attachmentSizes: AttachmentSizeMapType[],
+): ResultAsync<
+  AttachmentPresignedPostDataMapType[],
+  CreatePresignedPostError
+> => {
+  const totalAttachmentSizeLimit = fileSizeLimitBytes(FormResponseMode.Encrypt)
+  let totalAttachmentSize = 0
```

Comment:
minor /nit: would prefer to keep this as a pure function instead of having the side effect of mutating totalAttachmentSize

suggestion to do in two steps:

```
const totalAttachmentSize = attachmentSizes.map(({id}) => id).reduce((prev, next) => prev + next)
if (totalAttachmentSize > totalAttachmentSizeLimit) {...}

return ResultAsync.combine(...)

```

Reply:
Don't feel too strongly about this either, might be slightly more readable! Done in [0f80fa7](https://github.com/opengovsg/FormSG/pull/6685/commits/0f80fa72571d859ce883f9b5277065bf17c864d0)